### PR TITLE
Add nri-consul

### PR DESCRIPTION
**Pull request by Dentrax**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: nri-consul.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      repository: https://github.com/newrelic/nri-consul
+      expected-commit: 15b1cd07fadf7d2d45ff999e7c80fd066f738537
+      tag: v${{package.version}}
+
+  - uses: go/build
+    with:
+      modroot: .
+      packages: ./src/
+      output: nri-consul
+      ldflags: -s -w
```
Comment (ajayk) [inline nri-consul.yaml:29]:
```suggestion
      ldflags: -w
```

Comment (Dentrax) [reply nri-consul.yaml:29]: 🤔

File: nri-consul.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    - license: MIT
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - go
+      - build-base
+
```
Comment (imjasonh) [inline nri-consul.yaml:16]:
```suggestion

```

File: nri-consul.yaml (hunk 1 above)
Comment (ajayk) [reply nri-consul.yaml:29]: context here https://github.com/wolfi-dev/os/pull/6477

File: nri-consul.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package:
+  name: nri-consul
+  version: 2.7.2
+  epoch: 0
+  description: New Relic Infrastructure Consul Integration
+  copyright:
+    - license: MIT
+pipeline:
```
Comment (imjasonh) [inline nri-consul.yaml:8]:
```suggestion

  pipeline:
```

File: nri-consul.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package:
+  name: nri-consul
+  version: 2.7.2
+  epoch: 0
+  description: New Relic Infrastructure Consul Integration
+  copyright:
+    - license: MIT
+
+  pipeline:
```
Comment (imjasonh) [inline nri-consul.yaml:9]:
```suggestion
pipeline:
```

my bad

**Review by imjasonh — APPROVED** [review]

**Pull request merged**